mono/mini/mini-trampolines.c

```diff
@@ -822,16 +813,21 @@ common_call_trampoline (mgreg_t *regs, guint8 *code, MonoMethod *m, MonoVTable *
 gpointer
 mono_magic_trampoline (mgreg_t *regs, guint8 *code, gpointer arg, guint8* tramp)
 {
-	MonoError error;
 	gpointer res;
 
+	MONO_PREPARE_RESET_BLOCKING_UNBALANCED;
+
+	MonoError error;
+
 	trampoline_calls ++;
 
 	res = common_call_trampoline (regs, code, (MonoMethod *)arg, NULL, NULL, &error);
-	if (!mono_error_ok (&error)) {
-		mono_error_set_pending_exception (&error);
-		return NULL;
-	}
+	mono_error_raise_exception (&error);
```

Comment:
@vargaz this is a special case, as trampolines can't 100% depend on the caller checking for pending exception.